### DISCUSSION TITLE
"ValueError: <class 'src.models.calls.Call'> has no matching SQLAlchemy type" when adding Relationship

### DISCUSSION THREAD

**Discussion by GigaDroid** [fastapi/sqlmodel#908] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field, Relationship, SQLModel
from datetime import datetime
from uuid import UUID, uuid4

class Call(SQLModel, table=True):
    id: UUID | None = Field(default_factory=uuid4, primary_key=True)
    messages: list["Transcription"] = Relationship(back_populates="call")


### TRANSCRIPT ###

class BaseTranscription(SQLModel):
    id: UUID | None = Field(default_factory=uuid4, primary_key=True)
    message: str
    timestamp: datetime
    is_remote: bool
    is_final: bool
    call_id: UUID = Field(foreign_key="call.id")
    call: Call = Relationship(back_populates="messages")

class Transcription(BaseTranscription, table=True):
    pass
```

### Description

Hey, I have the following classes as shown in the example code. If I run this I get the following error:
ValueError: <class 'src.models.calls.Call'> has no matching SQLAlchemy type

However, when I run this after removing "call: Call = Relationship(back_populates="messages")" from the BaseTranscription class it actually works. I'm folllowing the documentation (https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/back-populates/#fix-it-using-back_populates) and it should be the same so I don't see why this doesn't work.

### Operating System

macOS

### SQLModel Version

0.0.16

### Python Version

3.12.2

**Comment by joachimhuet** [comment]
`class BaseTranscription(SQLModel):` is not a table add `table=True` for it to be recognized as a table, if the table is actually the `Transcription` class move there the relationship:
```python
class BaseTranscription(SQLModel):
    call: Call # optional if you don't need it on your base class

class BaseTranscription(BaseTranscription, table=True):
    call: Call = Relationship(back_populates="messages") # relationships should be defined on tables only
```

**Comment by GigaDroid** [comment]
@joachimhuet I'm sorry I forgot the last part of the example code. Here it is:
`class Transcription(BaseTranscription, table=True):
    pass`

**Reply by joachimhuet** [reply, marked as answer]
Well then, as I stated in my answer you can move the relationship in the `Transcription` class and it will work. You can still keep `call: Call`in the base class but just without the `= Relationship(...)` part. 😸
